app/controllers/front_controller.rb

```diff
@@ -0,0 +1,96 @@
+class FrontController < ActionController::Metal
+
+  def call
+    # retrieve the artefact from content-api
+    edition = edition_from_content_api(params[:slug], params[:edition])
+
+    if edition
+      # memoize the artefact
+      request.env[:__edition] = edition
+    else
+      # cacheable 404
+      set_expiry(10.minutes)
+      error 404
+      return
+    end
+
+    # set the params hash to the expected values
+    # not sure if this is useful
+    request.params[:controller] = format_controller(edition.format).class.to_s.downcase.gsub('controller', '')
+    request.params[:action] = action_from_request_method(request.method).to_s
+
+    # delegate this request to the appropriate controller
+    status, headers, body = format_controller(edition.format).dispatch(
+      action_from_request_method(request.method),
+      request
+    )
```

Comment:
The `MyController#dispatch` method is not documented API - the method has changed signature from Rails 4.2 to Rails 5.0. I would recommend not using this private API.